data/crac/crac-impl/src/main/java/com/farao_community/farao/data/crac_impl/SimpleCrac.java

```diff
@@ -337,14 +337,8 @@ public Cnec addCnec(String id, String networkElementId, AbstractThreshold abstra
     @Override
     public void addCnec(Cnec cnec) {
         addState(cnec.getState());
-        NetworkElement networkElement = addNetworkElement(cnec.getNetworkElement());
-        cnecs.add(new SimpleCnec(
-            cnec.getId(),
-            cnec.getName(),
-            networkElement,
-            (AbstractThreshold) cnec.getThreshold(),
-            getState(cnec.getState().getId())
-        ));
+        addNetworkElement(cnec.getNetworkElement());
+        cnecs.add(((SimpleCnec) cnec).copy(getState(cnec.getState().getId())));
```

Comment:
When you copy you're not sure that the network element instance in the cnec is present in crac set